### PR TITLE
Add additional preds checks to flowgraph.

### PR DESCRIPTION
**Pull request by sandreenko**

### CODE REVIEW

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Windows_NT x86 Checked jitstress2
@dotnet-bot test Windows_NT x64 Checked r2r_jitstress2

File: src/jit/flowgraph.cpp [hunk 1]
```diff
@@ -20637,6 +20646,12 @@ unsigned BBPredsChecker::CheckBBPreds(BasicBlock* block, unsigned curTraversalSt
     }
 
     unsigned blockRefs = 0;
+
+    CompAllocator memAllocator(comp, CMK_DebugOnly);
+
+    typedef JitHashTable<unsigned, JitSmallPrimitiveKeyFuncs<unsigned>, bool> VisitedPreds;
```
Comment (mikedn) [inline src/jit/flowgraph.cpp]: Hmm, maybe I should make a proper `JitHashSet`... There are other places that use this workaround, including non debug code.

**Comment by sandreenko** [comment]
Example of errors revealed with these checks: [x86 JitStress2](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x86_checked_windows_nt_jitstress2_prtest/79/), [x64 JitStress2](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_jitstress2_prtest/98/).

**Comment by sandreenko** [comment]
I will reopen this PR when we fix these errors.

**Pull request closed without merging**